### PR TITLE
Fat 2229/c455

### PR DESCRIPTION
**Pull request by kimkate303**

### CODE REVIEW

**Comment by kimkate303** [comment]
https://foliotest.testrail.io/index.php?/cases/view/455 - test case

File: cypress/integration/fee-fine/newFeeFines.spec.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import uuid from 'uuid';
+import moment from 'moment';
 import TestType from '../../support/dictionary/testTypes';
 import Features from '../../support/dictionary/features';
 import PatronGroups from '../../support/fragments/settings/users/patronGroups';
 import getRandomPostfix from '../../support/utils/stringTools';
```
Comment (MaksimTerekhov) [inline cypress/integration/fee-fine/newFeeFines.spec.js:6]: 'getRandomPostfix' is defined but never used.

File: cypress/integration/fee-fine/newFeeFines.spec.js [hunk 2]
```diff
@@ -11,107 +13,190 @@ import UsersCard from '../../support/fragments/users/usersCard';
...
-    PatronGroups.createViaApi(patronGroupName).then(patronGroupId => {
+    PatronGroups.createViaApi().then(patronGroupId => {
       testData.patronGroupId = patronGroupId;
       testData.userProperties = { barcode : uuid() };
-      Users.createViaApi({ ...Users.defaultUser, patronGroup: patronGroupId, barcode: testData.userProperties.barcode }).then(userProperties => {
+      Users.createViaApi({
+        patronGroup: patronGroupId,
+        ...userData
+      }).then(userProperties => {
+        console.log(userProperties);
```
Comment (MaksimTerekhov) [inline cypress/integration/fee-fine/newFeeFines.spec.js:40]: remove debug info pls

File: cypress/integration/fee-fine/newFeeFines.spec.js [hunk 3]
```diff
@@ -11,107 +13,190 @@ import UsersCard from '../../support/fragments/users/usersCard';
...
   });
 
   it('C455 Verify "New fee/fine" behavior when "Charge & pay now" button pressed', { tags: [TestType.smoke, Features.feeFine] }, () => {
-    const initialCheckNewFeeFineFragment = (ownerName = '') => {
-      NewFeeFine.checkInitialState(testData.userProperties, ownerName);
+    const feeInfo = [testData.owner.name, testData.feeFineType.feeFineTypeName, 'Paid fully'];
+    const itemInfo = [testData.instanceTitle + ' (book)', ITEM_BARCODE];
+
+    const initialCheckNewFeeFineFragment = () => {
+      console.log(testData.userProperties);
```
Comment (MaksimTerekhov) [inline cypress/integration/fee-fine/newFeeFines.spec.js:106]: remove debug info pls

File: cypress/support/fragments/users/newFeeFine.js [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
     cy.expect(rootModal.find(Button(`${userLastName}, ${userFirstName} ${userMiddleName}`)).exists());
     cy.expect(rootModal.find(Button(userBarcode)).exists());
-    cy.expect(feefineOwnerSelect.has({ value:ownerName }));
+    cy.expect(Option(ownerName).exists());
     cy.expect(feeFineTypeSelect.has({ value:'' }));
     cy.expect(rootModal.find(TextField({ id:'amount' })).has({ value: '' }));
     cy.expect(rootModal.find(TextField({ placeholder:'Scan or enter item barcode' })).has({ value: '' }));
     cy.expect(rootModal.find(TextArea({ id:'comments' })).has({ value:'' }));
   },
+  checkClosedFees(feeProps, index) {
```
Comment (MaksimTerekhov) [inline cypress/support/fragments/users/newFeeFine.js:20]: better to use name like _checkClosedFeesByRow_

File: cypress/integration/fee-fine/newFeeFines.spec.js [hunk 5]
```diff
@@ -11,108 +12,186 @@ import UsersCard from '../../support/fragments/users/usersCard';
...
   it('C455 Verify "New fee/fine" behavior when "Charge & pay now" button pressed (spitfire)', { tags: [TestType.smoke, Features.feeFine, devTeams.firebird] }, () => {
-    const initialCheckNewFeeFineFragment = (ownerName = '') => {
-      NewFeeFine.checkInitialState(testData.userProperties, ownerName);
+    const feeInfo = [testData.owner.name, testData.feeFineType.feeFineTypeName, 'Paid fully'];
+    const itemInfo = [testData.instanceTitle + ' (book)', ITEM_BARCODE];
+    const initialCheckNewFeeFineFragment = () => {
+      NewFeeFine.checkInitialState(testData.userProperties, testData.owner.name);
       NewFeeFine.setFeeFineOwner(testData.owner.name);
-      NewFeeFine.checkFilteredFeeFineType(testData.feeFineType.feeFineTypeName);
+      NewFeeFine.checkFilteredFeeFineType(testData.feeFineType.feeFineTypeName);
```
Comment (MaksimTerekhov) [inline cypress/integration/fee-fine/newFeeFines.spec.js:101]: image

File: cypress/integration/fee-fine/newFeeFines.spec.js [hunk 6]
```diff
@@ -11,108 +12,186 @@ import UsersCard from '../../support/fragments/users/usersCard';
...
+                  materialType: { id: testData.materialType },
+                }],
+              ],
+            });
           });
       });
     });
   });
 
   it('C455 Verify "New fee/fine" behavior when "Charge & pay now" button pressed (spitfire)', { tags: [TestType.smoke, Features.feeFine, devTeams.firebird] }, () => {
```
Comment (MaksimTerekhov) [inline cypress/integration/fee-fine/newFeeFines.spec.js:95]: change devTeams to spitfire please
image

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/integration/fee-fine/newFeeFines.spec.js (hunk 5 above)
Comment (kimkate303) [reply cypress/integration/fee-fine/newFeeFines.spec.js]: done

File: cypress/integration/fee-fine/newFeeFines.spec.js (hunk 6 above)
Comment (kimkate303) [reply cypress/integration/fee-fine/newFeeFines.spec.js]: done

File: cypress/integration/fee-fine/newFeeFines.spec.js [hunk 7]
```diff
@@ -11,108 +12,186 @@ import UsersCard from '../../support/fragments/users/usersCard';
...
+import NewFeeFine from '../../support/fragments/users/newFeeFine';
 import devTeams from '../../support/dictionary/devTeams';
 
 describe('Fee/fine management', () => {
-  const testData = {};
+  const testData = {
+    instanceTitle: `Pre-checkin_instance_${Number(new Date())}`
+  };
+  const userData = { ...DefaultUser.defaultApiPatron };
+  const ITEM_BARCODE = generateItemBarcode();
```
[the remark this answers is not in the document]
Comment (kimkate303) [reply cypress/integration/fee-fine/newFeeFines.spec.js:30]: may be even itemBarcode?

[the remark this answers is not in the document]
Comment (penfriend) [reply cypress/integration/fee-fine/newFeeFines.spec.js:30]: Agree

**Review by penfriend — APPROVED** [review]

**Pull request merged**